test/github-issues/3375/issue-3375.ts

```diff
@@ -0,0 +1,141 @@
+            const migrationExecutor = new MigrationExecutor2(connection, queryRunner);
+            const executedMigrations = await migrationExecutor.loadExecutedMigrations(queryRunner);
+            expect(executedMigrations.length).to.equal(1);
+            executedMigrations.forEach(executedMigration => {
+                expect(executedMigration.name.length).to.be.greaterThan(0);
+                expect(executedMigration.hash.length).to.be.greaterThan(0);
+            });
+        })));
+    });
+
+    describe("existing migration table", () => {
+        setup();
+
+        it("should add hash column when it does not exist and calculate hashes of executed migrations", () => Promise.all(connections.map(async connection => {
+            await connection.runMigrations();
+            const queryRunner = connection.createQueryRunner();
+            if (connection.driver instanceof MongoDriver) {
+                const mongoRunner = queryRunner as MongoQueryRunner;
+                await mongoRunner.databaseConnection.db(connection.driver.database!).collection("migrations").updateMany({}, {$unset: {hash: 1}});
+            } else {
+                await queryRunner.dropColumn("migrations", "hash");
+            }
+
+            await connection.runMigrations();
+            const migrationExecutor = new MigrationExecutor2(connection, queryRunner);
+            const executedMigrations = await migrationExecutor.loadExecutedMigrations(queryRunner);
+            expect(executedMigrations.length).to.equal(1);
+            executedMigrations.forEach(executedMigration => {
+                expect(executedMigration.name.length).to.be.greaterThan(0);
+                expect(executedMigration.hash.length).to.be.greaterThan(0);
+            });
+        })));
+
+    });
+
+    describe("migration hash mismatch", () => {
+
+        beforeEach(async () => {
+            connections = await createConnections();
+            await Promise.all(connections.map(connection => connection.runMigrations()));
+            await closeTestingConnections(connections);
+        });
+
+        setup(__dirname + "/migration-invalid-hash/*.js", false);
+
+        afterEach(() => Promise.all(connections.map(async connection => {
+            delete (connection.options as any).migrationsForce;
+        })));
+
+        it("should fail when migration hash differs", () => Promise.all(connections.map(async connection => {
+            (connection.options as any).migrationsForce = undefined;
+            let error: Error|undefined = undefined;
+            try {
+                await connection.runMigrations();
+            } catch (err) {
+                error = err;
+            }
+            expect(error).to.be.ok;
+            expect(error!.message).to.match(/migration hash/i);
+        })));
+
+        it("should succeed when migrationIgnoreHash is truthy", () => Promise.all(connections.map(async connection => {
```

Comment:
migrationsForce instead of migrationIgnoreHash?

Reply:
True, this was an old name of the variable. Changed it!